### PR TITLE
feat: Add name validation for namespaces, components, and endpoints

### PR DESCRIPTION
**Pull request by jthomson04**
#### What does the PR do?
Adds name validation for namespaces, components, and endpoints. Also enable some disabled tests which rely on this functionality.

#### Checklist
- [ ] PR title reflects the change and is of format `<commit_type>: <Title>`
- [ ] Changes are described in the pull request.
- [ ] Related issues are referenced.
- [ ] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [ ] Added [test plan](#test-plan) and verified test passes.
- [ ] Verified that the PR passes existing CI.
- [ ] Verified copyright is correct on all changed files.
- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [ ] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [ ] build
- [ ] ci
- [ ] docs
- [x ] feat
- [ ] fix
- [ ] perf
- [ ] refactor
- [ ] revert
- [ ] style
- [ ] test

#### Where should the reviewer start?
See runtime/rust/component.rs for validation code and unit tests demonstrating valid and invalid strings.

#### Test plan:

- CI Pipeline ID:

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)
- closes GitHub issue: #xxx

### CODE REVIEW

**Comment by rmccorm4** [comment]
Hi @jthomson04, thanks for contributing! Can you sign your commits per https://github.com/triton-inference-server/triton_distributed/blob/main/CONTRIBUTING.md#developer-certificate-of-origin?

@nv-anants we need to address the workflow of validating external contributions from forks - our current gitlab triggers don't work for forks.

File: runtime/rust/src/component.rs [hunk 1]
```diff
@@ -233,105 +248,105 @@ fn validate_allowed_chars(input: &str) -> Result<(), ValidationError> {
...
-//     //         let name_errors = &error_map["name"];
-//     //         assert_eq!(name_errors[0].code, "invalid_characters");
-//     //     }
-//     // }
-
-//     #[test]
-//     fn test_edge_cases() {
-//         // Edge cases
-//         let edge_inputs = vec![
-//             ("-", true),   // Single hyphen
```
Comment (rmccorm4) [inline runtime/rust/src/component.rs:320]: @ryanolson placeholder test allowed pre/postfix hyphens, new changes disallow it - which one did you want?

**Review by rmccorm4 — COMMENTED** [review]
Code changes LGTM! 

Just a couple small things to address:
- the CI checks: https://github.com/triton-inference-server/triton_distributed/pull/127#issuecomment-2646764097
- double checking with @ryanolson on expected [regex validation criteria](https://github.com/triton-inference-server/triton_distributed/pull/127#discussion_r1948317053)

File: runtime/rust/src/component.rs [hunk 2]
```diff
@@ -200,31 +205,41 @@ pub struct Namespace {
...
+
+        Ok(component)
     }
 }
 
 // Custom validator function
 fn validate_allowed_chars(input: &str) -> Result<(), ValidationError> {
     // Define the allowed character set using a regex
-    let regex = regex::Regex::new(r"^[a-z0-9-_]+$").unwrap();
+    let regex = regex::Regex::new(r"^[a-z0-9_]+(?:-+[a-z0-9_]+)*$").unwrap();
```
Comment (rmccorm4) [inline runtime/rust/src/component.rs:242]: What is defining or backing the validation criteria here? Are there any dependent/third-party sources or conventions that require a certain format or that we're following?

Comment (ryanolson) [reply runtime/rust/src/component.rs:241]: primary what is allowed for nats subjects

File: runtime/rust/src/component.rs (hunk 1 above)
Comment (ryanolson) [reply runtime/rust/src/component.rs:322]: names must begin with a letter.  maybe alphanumeric, open to that.  no leading or trailing symbols for the public api.  we might consider allowing leading under scores for things on the internal api which would then guarantee no collisions
